scripts/approximations/polynomial.py

```diff
@@ -0,0 +1,41 @@
+
+
+import numpy as np
+
+def construct_vandermonde_matrix(x_list: list) -> list[list]:
+    """ Constructs a Vandermonde matrix for a polynomial approximation.
+    from the list of x values given.
+    
+    len(x_list) * len(x_list)
+    x_list is the list of all x values to construct the matrix from.
+    
+    V = [1 x_0 x_0^2 ... x_0^{n-1}]
+    [1 x_1 x_2^1 ... x_1^{n-1}]
+    ...
+    [1 x_n x_n^2 ... x_n^{n-1}]
+
+    Vandermonde matrix is a matrix with the terms of a geometric progression in each row.
+    We use Vandermonde matrix to convert the system of linear equations into a linear algebra problem
+    that we can solve.
+    """
+    num_terms = len(x_list)
+
+    matrix = []
+
+    for i in range(num_terms):
+        row = []
+        for j in range(num_terms):
+            row.append(x_list[i]**j)
+        matrix.append(row)
+
+    return matrix
+
+def solve(x: np.ndarray, coeffs: np.ndarray) -> np.ndarray:
+    """ Solves the polynomial. Given a list of x coordinates and a list of coefficients, returns a list of
```

Comment:
This should probably be renamed `evaluate_polynomial`, were not solving the system of equations in this fn

Reply:
I'm generally confused. Why don't we just do this as:

![image](https://user-images.githubusercontent.com/6440154/205805374-77796913-7ab0-4d3f-8060-6bb35adb1b8d.png)

Reply:
(you can change np.dot to just be matrix_inv @ y)

Reply:
We are solving for the coefficients that are unknown. x and y are both known in this system of equations and we treat them as constants.

Reply:
Resolving this since seeing an ACK. The function name has been changed.